akka-docs-dev/rst/scala/code/docs/stream/StreamPartialFlowGraphDocSpec.scala

```diff
@@ -142,4 +144,62 @@ class StreamPartialFlowGraphDocSpec extends AkkaSpec {
 
     Await.result(matSink, 300.millis) should equal(1 -> "1")
   }
+
+  "help users in providing complex ports" in {
+    object AudioPorts extends Ports {
+      val nums = Input[Int] // TODO: use I / O words or keep sinks/sources?
+      val words = Input[String]
+
+      val out = Output[String]
+    }
+
+    // building
+    val g = PartialFlowGraph(AudioPorts) { implicit b ⇒
```

Comment:
Interesting idea / spin on the above (a bit akka-http routing inspired):

```scala
    val g = PartialFlowGraph(AudioPorts) { implicit b ⇒ ports ⇒
        import FlowGraphImplicits._
        val merge = Merge[String]
      // format: OFF
      ports.nums ~> Flow[Int].map(_.toString) ~> merge
                                  ports.words ~> merge ~> ports.out
      // format: ON
    }
```

yay or nay? Might be "too fancy"